pinot-core/src/test/java/org/apache/pinot/queries/ArgMinMaxTest.java

```diff
@@ -0,0 +1,613 @@
+        + "argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn), "
+        + "argmin(replace(stringColumn, \'a\', \'bb\'), replace(stringColumn, \'a\', \'bb\'))"
+        + "FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    assertEquals(rows.get(0)[0], 7996000D);
+    assertEquals(rows.get(0)[1], 1500D);
+    assertEquals(rows.get(0)[2], 2250000D);
+    assertEquals(rows.get(0)[3], "bb11");
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 1500D);
+    assertEquals(rows.get(1)[2], 2250000D);
+    assertEquals(rows.get(1)[3], "bb11");
+    assertNull(rows.get(2)[0]);
+    assertNull(rows.get(2)[1]);
+    assertEquals(rows.get(2)[3], "bb11");
+    assertNull(rows.get(3)[0]);
+    assertNull(rows.get(3)[1]);
+    assertEquals(rows.get(3)[3], "bb11");
+
+    // Inter segment mix aggregation function with CASE statement
+    query = "SELECT argmin(CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' ELSE 'c' END"
+        + ", stringColumn), argmin(CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' "
+        + "ELSE 'c' END, CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' ELSE 'c' END) "
+        + "FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    for (int i = 0; i < 4; i++) {
+      assertEquals(rows.get(i)[0], "a22");
+      assertEquals(rows.get(i)[1], "a");
+    }
+
+    // TODO: The following query throws an exception,
+    //       requires fix for multi-value bytes column serialization in DataBlock
+    query = "SELECT arg_min(intColumn, mvBytesColumn) FROM testTable";
+
+    try {
+      brokerResponse = getBrokerResponse(query);
+      fail("remove this test case, now mvBytesColumn works correctly in serialization");
+    } catch (Exception e) {
+      assertTrue(e.getMessage()
+          .contains("java.lang.IllegalArgumentException: Unsupported type of value: byte[][]"));
+    }
+  }
+
+  @Test
+  public void testAggregationDedupe() {
+    // Inter segment dedupe test1 without dedupe
+    String query = "SELECT  "
+        + "argmin(booleanColumn, bigDecimalColumn, intColumn) FROM testTable WHERE doubleColumn <= 1200";
+
+    BrokerResponseNative brokerResponse = getBrokerResponse(query);
+    ResultTable resultTable = brokerResponse.getResultTable();
+    List<Object[]> rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    assertEquals(rows.get(0)[0], 0);
+    assertEquals(rows.get(1)[0], 1200);
+    assertEquals(rows.get(2)[0], 0);
+    assertEquals(rows.get(3)[0], 1200);
+
+    // test1, with dedupe
+    query = "SELECT  "
```

Comment:
Shall we have a query option or system option to bypass the default behavior and instead return just one of the rows for all duplicate / matching rows where min / max is happening ?

Reply:
We can maybe do this as a follow up. It will require the stability sorting/dedup of projection results, as would be better to do in a seperate PR.